### ISSUE TITLE
Can't use AMD rocm with kernel 5.8.0 (installed by linux-system76)

### ISSUE THREAD

**Issue by apalazzi** [pop-os/pop#1402]
**Distribution (run `cat /etc/os-release`):**

NAME="Pop!_OS"
VERSION="20.04 LTS"
ID=ubuntu
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 20.04 LTS"
VERSION_ID="20.04"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
LOGO=distributor-logo-pop-os
 
**Related Application and/or Package Version (run `apt policy $PACKAGE NAME`):**
rocm-dkms:
  Installed: 3.9.0.30900-17
  Candidate: 3.9.0.30900-17
  Version table:
 *** 3.9.0.30900-17 500
        500 http://repo.radeon.com/rocm/apt/debian xenial/main amd64 Packages
        100 /var/lib/dpkg/status

**Issue/Bug Description:**
Package linux-system76 depends on kernel 5.8.0, however this kernel is not supported by AMD's computation stack rocm https://github.com/RadeonOpenCompute and the installation of rocm-dkms ends with an error:

```Building initial module for 5.8.0-7630-generic
ERROR (dkms apport): kernel package linux-headers-5.8.0-7630-generic is not supported
Error! Bad return status for module build on kernel: 5.8.0-7630-generic (x86_64)
[...]
dpkg: errore nell'elaborare il pacchetto rocm-dkms (--configure):
 problemi con le dipendenze - lasciato non configurato
Si sono verificati degli errori nell'elaborazione:
 rock-dkms
 rocm-dkms
```

**Steps to reproduce (if you know):**
1. Update to latest pop os version: apt update&&apt upgrade
1. Install rocm as per the instructions https://rocmdocs.amd.com/en/latest/Installation_Guide/Installation-Guide.html#installing-a-rocm-package-from-a-debian-repository

**Expected behavior:**
One should be able to use AMD's rocm without removing a 'required' package - pop-desktop depends on linux-system76

**Other Notes:**
Workaround: apt remove linux-system76 linux-image-5.8.0-7630-generic

**Comment by jacobgkau** [comment]
Part of that error message is not in English, but it looks like `rock-dkms` failed to configure, which may have caused `rocm-dkms` to also fail to configure. I see reports [here](https://github.com/RadeonOpenCompute/ROCm/issues/1263) that ROCm should work with kernel 5.8, but I do see [here](https://github.com/RadeonOpenCompute/ROCK-Kernel-Driver/issues/94) that ROCK doesn't work with kernel versions above 5.7, and also that it hardcodes a list of acceptable OS's (similar to AMDGPU-PRO, see [here](https://github.com/pop-os/pop/issues/1135)). Both of those things would need to be fixed in `rock-dkms`.

From my understanding, the dkms packages contain things that AMD is trying to upstream to the Linux kernel anyway. From your rocmdocs link, see this paragraph:

> Note: These directions may not work as written on unsupported Debian-based distributions. For example, newer versions of Ubuntu may not be compatible with the rock-dkms kernel driver. In this case, you can exclude the rocm-dkms and rock-dkms packages.

So it may be possible to use ROCm without those DKMS packages, using the ROCm drivers included in Linux.

**Comment by apalazzi** [comment]
Thanks for the info; I didn't find the (closed) issue on Ubuntu 20.10, and in fact it seems that the rocm stack works also without the rocm-dkms and rock-dkms packages, at least on the 5.8.0 kernel.

I'll do some more testing, but I think the issue can be closed.
